core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -647,6 +651,30 @@ public boolean canDeleteIndexContents(Index index, IndexSettings indexSettings,
     }
 
     /**
+     * Verify that the contents on disk for the given index is deleted; if not, delete the contents if allowed.
+     * @param index {@code Index} to make sure its deleted from disk
+     * @param clusterState the current {@code ClusterState}
+     * @return true if the index contents is deleted on this node
+     */
+    public boolean tryDeleteIndexContents(final Index index, final ClusterState clusterState) {
+        boolean success = false;
+        if (canDeleteIndexContents(index, null, false)) {
+            try {
+                final IndexMetaData metaData = metaStateService.loadIndexState(index);
+                deleteClosedIndex("deleted index was not assigned to local node", metaData, clusterState);
+                success = true;
+            } catch (IOException e) {
+               success = false;
+            }
+        }
+        if (success == false) {
+            logger.warn("[{}] Deletion request unsuccessful, likely scenario is an index was " +
+                        "created and deleted while the node was offline.", index);
+        }
+        return success;
+    }
```

Comment:
I think this cleans it up nicely? https://github.com/bleskes/elasticsearch/commit/9225b3e1c3c50530b713cda63fa2084e03a68c11   (changed some naming ) WDYT?